pkg/api/validate/types.go

```diff
@@ -17,6 +17,10 @@ var (
 		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
 		`$`)
 
+	rxRfc3986 = regexp.MustCompile(`^(([^:/?#]+):)?` +
```

Comment:
OK, let's not go here.  How about `if _, err := url.Parse(foo); err != nil` in the caller?